src/middlewared/middlewared/plugins/disk.py

```diff
@@ -1061,6 +1101,61 @@ def label_to_dev(self, label, geom_scan=True):
 
         await self.sync(dev)
 
+    @private
+    def format(self, disk, swapgb):
+
+        job = self.middleware.call_sync('disk.wipe', disk, 'QUICK')
+        job.wait_sync()
+        if job.error:
+            raise CallError(f'Failed to wipe disk {disk}: {job.error}')
+
+        # Calculate swap size.
+        swapsize = swapgb * 1024 * 1024 * 2
+        # Round up to nearest whole integral multiple of 128 and subtract by 34
+        # so next partition starts at mutiple of 128.
+        swapsize = (int((swapsize + 127) / 128)) * 128
+
+        geom.scan()
+        g = geom.geom_by_name('DISK', disk)
+        if g and g.provider.mediasize:
+            size = g.provider.mediasize
+            # The GPT header takes about 34KB + alignment, round it to 100
+            if size - 100 <= swapgb * 1024 * 1024:
+                raise CallError(f'Your disk size must be higher than {swapgb}GB')
```

Comment:
It is possible to check this before wiping disk. I think we should do it, there is a tiny-tiny change it may save someone's data.

Reply:
Good call, fixed

Reply:
Done